src/herbivoresAndCarnivores.js

```diff
@@ -1,15 +1,60 @@
 'use strict';
 
 class Animal {
-  // write your code here
+  static alive = [];
+
+  constructor(name, health = 100) {
+    this.name = name;
+    this.health = health;
+  }
 }
 
 class Herbivore extends Animal {
-  // write your code here
+  hidden = false;
+
+  constructor(name, health = 100) {
+    super(name, health);
+
+    this.name = name;
+    this.health = health;
+
+    Animal.alive.push(this);
+  }
+
+  hide() {
+    return (this.hidden = true);
+  }
 }
 
 class Carnivore extends Animal {
-  // write your code here
+  constructor(name, health) {
+    super(name, health);
+    Animal.alive.push(this);
+  }
+
+  bite(animal) {
+    if (animal instanceof Carnivore) {
+      return;
+    }
+
+    if (animal.hidden === true) {
+      return;
+    }
+
+    animal.health -= 50;
+
+    if (animal.health === 0) {
+      const AnimalAliveNewList = [];
+
+      Animal.alive.map((item) => {
+        if (item.health > 0) {
+          AnimalAliveNewList.push(item);
+        }
+      });
+
+      Animal.alive = AnimalAliveNewList;
```

Comment:
[CODE PERFORMANCE] - The current implementation for removing dead animals from the `Animal.alive` array is not optimal. You should use `Array.prototype.filter` instead of `map` and a separate array to avoid nesting loops. Replace lines 47-55 with something like `Animal.alive = Animal.alive.filter(item => item.health > 0);`.